contracts/contracts/harvest/Harvester.sol

```diff
@@ -0,0 +1,376 @@
+// SPDX-License-Identifier: agpl-3.0
+pragma solidity ^0.8.0;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { SafeMath } from "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "@openzeppelin/contracts/utils/math/Math.sol";
+
+import { StableMath } from "../utils/StableMath.sol";
+import { Initializable } from "../utils/Initializable.sol";
+import { Governable } from "../governance/Governable.sol";
+import { IVault } from "../interfaces/IVault.sol";
+import { IOracle } from "../interfaces/IOracle.sol";
+import { IStrategy } from "../interfaces/IStrategy.sol";
+import { IUniswapV2Router } from "../interfaces/uniswap/IUniswapV2Router02.sol";
+import "../utils/Helpers.sol";
+
+contract Harvester is Initializable, Governable {
+    using SafeERC20 for IERC20;
+    using SafeMath for uint256;
+    using StableMath for uint256;
+
+    event UniswapUpdated(address _address);
+    event SwapTokenAdded(address _address);
+    event SwapTokenRemoved(address _address);
+    event RewardTokenConfigUpdated(
+        address _tokenAddress,
+        uint32 _allowedSlippageBps,
+        uint32 _harvestRewardBps,
+        address _uniswapV2CompatibleAddr,
+        uint256 _liquidationLimit
+    );
+
+    // Tokens that should be swapped for stablecoins
+    address[] public swapTokens;
+
+    // Strategies approved for use by the Vault
+    struct RewardTokenConfig {
+        // Max allowed slippage when swapping reward token for a stablecoin denominated in basis points.
+        uint32 allowedSlippageBps;
```

Comment:
Since bps maxes out at 10,000 = 100%, we could change these to uint16s for storage, which hold up to 65,536.

Reply:
good point thanks, changing to uint16